### ISSUE TITLE
diffusion policy: a very strange train loss

### ISSUE THREAD

**Issue by flyingTeng** [huggingface/lerobot#2796]
### Ticket Type

❓ Technical Question

### Environment & System Info

```Shell
Copy-and-paste the text below in your GitHub issue and FILL OUT the last point.

- lerobot version: 0.4.1
- Platform: Linux-5.15.0-131-generic-x86_64-with-glibc2.31
- Python version: 3.10.19
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- PyTorch version: 2.7.1+cu126
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.6
- GPU model: NVIDIA GeForce RTX 3090
- Using GPU in script?: <fill in>
```

### Description

<img width="812" height="292" alt="Image" src="https://github.com/user-attachments/assets/c84af1eb-fd5b-49de-9610-c050af3f55f5" />
The same dataset performs well on ACT.
At the same time, the training speed of diffusion is terrible.like https://github.com/huggingface/lerobot/issues/1488
and https://github.com/huggingface/lerobot/issues/2569

### Context & Reproduction

`{
  "dataset": {
    "repo_id": "local/aligned_alidata",
    "root": "/home/li/tzf/lerobot/final_merged_dataset",
    "use_imagenet_stats": true,
    "image_transforms": {
        "enable": true
    }
  },
  "policy": {
    "type": "diffusion",
    "device": "cuda",
    "use_amp": true,
    "push_to_hub": false,
    "n_obs_steps": 2,
    "horizon": 16,
    "n_action_steps": 8,
    "vision_backbone": "resnet18",
    "pretrained_backbone_weights": "ResNet18_Weights.IMAGENET1K_V1",
    "optimizer_lr": 1e-4,
    "optimizer_lr_backbone": 1e-5,
    "optimizer_weight_decay": 1e-6,
    "optimizer_weight_decay_backbone": 1e-6,
    "use_separate_rgb_encoder_per_camera" : true,
    "use_group_norm": false,
    "crop_shape": [432, 576],
    "crop_is_random": true,
    "num_inference_steps": 100,
    "beta_schedule": "squaredcos_cap_v2",
    "prediction_type": "epsilon",
    "num_train_timesteps": 100,
    "clip_sample": false
  },
  "batch_size": 24,
  "steps": 50000,
  "eval_freq": 0,
  "log_freq": 1,
  "save_checkpoint": true,
  "save_freq": 2000,
  "num_workers": 8,
  "wandb": {
    "enable": 1
  }
}
`

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

**Comment by flyingTeng** [comment]
```
{
    "codebase_version": "v3.0",
    "robot_type": "unknown",
    "total_episodes": 148,
    "total_frames": 56000,
    "total_tasks": 1,
    "chunks_size": 1000,
    "data_files_size_in_mb": 100,
    "video_files_size_in_mb": 500,
    "fps": 30,
    "splits": {
        "train": "0:148"
    },
    "data_path": "data/chunk-{chunk_index:03d}/file-{file_index:03d}.parquet",
    "video_path": null,
    "features": {
        "observation.images.wrist": {
            "dtype": "image",
            "shape": [
                3,
                480,
                640
            ],
            "names": [
                "c",
                "h",
                "w"
            ]
        },
        "observation.images.external": {
            "dtype": "image",
            "shape": [
                3,
                480,
                640
            ],
            "names": [
                "c",
                "h",
                "w"
            ]
        },
        "observation.state": {
            "dtype": "float32",
            "shape": [
                8
            ],
            "names": [
                "j1",
                "j2",
                "j3",
                "j4",
                "j5",
                "j6",
                "j7",
                "gripper"
            ]
        },
        "action": {
            "dtype": "float32",
            "shape": [
                8
            ],
            "names": [
                "j1",
                "j2",
                "j3",
                "j4",
                "j5",
                "j6",
                "j7",
                "gripper"
            ]
        },
        "timestamp": {
            "dtype": "float32",
            "shape": [
                1
            ],
            "names": null
        },
        "frame_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "episode_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "task_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        }
    }
}
```
my datasets


and The actual deployment of diffusion is also terrible; after multiple rounds of training, the visual layer almost disappears, which forced me to use a lower learning rate.

**Comment by xh-g** [comment]
I get same question! Diffusion policy seems to work much more terrible compared to another baseline algorithm - ACT.

**Comment by jashshah999** [comment]
When a dataset works with ACT but not Diffusion, it's often a subtle data quality issue that only manifests with certain architectures. `lerobot-doctor check your_dataset --checks actions,statistics,anomalies` can surface these. Also try `lerobot-doctor gate --policy diffusion` which validates diffusion-specific requirements.

https://github.com/jashshah999/lerobot-doctor
